platformio.ini

```diff
@@ -195,325 +173,295 @@ build_flags =
 large_partitions = tools/WLED_ESP32_8MB.csv
 extreme_partitions = tools/WLED_ESP32_16MB_9MB_FS.csv
+board_build.partitions = ${esp32.default_partitions}
 lib_deps =
-  https://github.com/lorol/LITTLEFS.git
-  https://github.com/pbolduc/AsyncTCP.git @ 1.2.0
   ${env.lib_deps}
+  https://github.com/pbolduc/AsyncTCP.git @ 1.2.0
 # additional build flags for audioreactive
-AR_build_flags = -D USERMOD_AUDIOREACTIVE
+AR_build_flags = -DUSERMOD_AUDIOREACTIVE
 AR_lib_deps = kosme/arduinoFFT @ 2.0.1
 
 [esp32_idf_V4]
 ;; experimental build environment for ESP32 using ESP-IDF 4.4.x / arduino-esp32 v2.0.5
-;; very similar to the normal ESP32 flags, but omitting Lorol LittleFS, as littlefs is included in the new framework already.
 ;;
 ;; please note that you can NOT update existing ESP32 installs with a "V4" build. Also updating by OTA will not work properly.
 ;; You need to completely erase your device (esptool erase_flash) first, then install the "V4" build from VSCode+platformio.
-platform = espressif32@ ~6.3.2
-platform_packages = platformio/framework-arduinoespressif32 @ 3.20009.0    ;; select arduino-esp32 v2.0.9 (arduino-esp32 2.0.10 thru 2.0.14 are buggy so avoid them)
-build_flags = -g
+extends = esp32
+build_flags =
+  ${esp32.build_flags}
   -Wshadow=compatible-local ;; emit warning in case a local variable "shadows" another local one
-  -DARDUINO_ARCH_ESP32 -DESP32
-  -D CONFIG_ASYNC_TCP_USE_WDT=0
+  -DESP32
   -DARDUINO_USB_CDC_ON_BOOT=0 ;; this flag is mandatory for "classic ESP32" when building with arduino-esp32 >=2.0.3
-lib_deps =
-  https://github.com/pbolduc/AsyncTCP.git @ 1.2.0
-  ${env.lib_deps}
 
 [esp32s2]
 ;; generic definitions for all ESP32-S2 boards
-platform = espressif32@ ~6.3.2
-platform_packages = platformio/framework-arduinoespressif32 @ 3.20009.0    ;; select arduino-esp32 v2.0.9 (arduino-esp32 2.0.10 thru 2.0.14 are buggy so avoid them)
-build_flags = -g
-  -DARDUINO_ARCH_ESP32
+extends = esp32
+build_flags =
+  ${esp32.build_flags}
   -DARDUINO_ARCH_ESP32S2
   -DCONFIG_IDF_TARGET_ESP32S2=1
-  -D CONFIG_ASYNC_TCP_USE_WDT=0
-  -DARDUINO_USB_MSC_ON_BOOT=0 -DARDUINO_USB_DFU_ON_BOOT=0
+  -DCONFIG_ASYNC_TCP_USE_WDT=0
+  -DARDUINO_USB_MSC_ON_BOOT=0
+  -DARDUINO_USB_DFU_ON_BOOT=0
   -DCO
   -DARDUINO_USB_MODE=0 ;; this flag is mandatory for ESP32-S2 !
   ;; please make sure that the following flags are properly set (to 0 or 1) by your board.json, or included in your custom platformio_override.ini entry:
   ;; ARDUINO_USB_CDC_ON_BOOT
-lib_deps =
-  https://github.com/pbolduc/AsyncTCP.git @ 1.2.0
-  ${env.lib_deps}
-
-[esp32c3]
-;; generic definitions for all ESP32-C3 boards
-platform = espressif32@ ~6.3.2
-platform_packages = platformio/framework-arduinoespressif32 @ 3.20009.0    ;; select arduino-esp32 v2.0.9 (arduino-esp32 2.0.10 thru 2.0.14 are buggy so avoid them)
-build_flags = -g
-  -DARDUINO_ARCH_ESP32
-  -DARDUINO_ARCH_ESP32C3
-  -DCONFIG_IDF_TARGET_ESP32C3=1
```

Comment:
Where did you put this line? (`-DCONFIG_IDF_TARGET_ESP32C3=1`)

Its critical for ensuring a proper build on -S3.

Reply:
It's in the `[esp32c3]` section obviously :)